src/app/components/UserErrorComponent.test.js

```diff
@@ -60,4 +61,14 @@ describe('UserErrorComponent', () => {
 
     expect(component.find('p').at(1).text()).toEqual(en.same_value);
   });
+
+  it('Returns user friendly message when the transaction is rejected', () => {
+    const component = mount(
+      <Provider store={store}>
+        <UserErrorComponent message="user rejected transaction bla bla bla" visible />
+      </Provider>,
+    );
+
+    expect(component.find('p').at(1).text()).toEqual(en.user_rejected_transaction);
```

Comment:
en. user_rejected_transaction is "The user rejected the transaction on the wallet"
How is this test success, since the text "user rejected transaction bla bla bla" is different?

Reply:
Because this is the sentence under test:

if (message && message.toLowerCase().startsWith(USER_REJECTED_TRANSACTION)) {

The key here is startsWith()

Reply:
A good approach that will also work for future error cases not currently handled is to splice the entire error string from the start to the first bracket and display only that part of the error message. This is how we handled it in the rollup wallet.

Reply:
I just changed to Cheta's approach suggestion. Please check again.